ssz/src/test/java/tech/pegasys/teku/ssz/schema/SszUnionSchemaTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ssz.schema;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class SszUnionSchemaTest extends SszSchemaTestBase {
+
+  public static Stream<SszUnionSchema<?>> testUnionSchemas() {
```

Comment:
Should we add a case that has something non-primitive as one of the selector options?

Reply:
Good idea! ae935656